src/actions/guides/frontend/internationalization.cr

```diff
@@ -0,0 +1,341 @@
+    mkdir config/locales
+    ```
+
+    then add a localization
+    ```
+    # config/locales/en.yml
+    en:
+      auth:
+        sign_in: "Sign-In"
+        sign_up: "Sign-up"
+        update_pwd: "Update Password"
+        reset_pwd_request: "Reset Password"
+      default:
+        page_name: Welcome
+      me:
+        email: Mail
+        profile: Profil
+        next: "Next, you may want to"
+        auth_guides: "Check out the authentication guides"
+        modify_page: "Modify this page"
+        after_signin: "Change where you go after sign in"
+    ```
+
+    ## **Step 3** - configure i18n to work with Lucky
+    ```
+    # config/i18n.cr
+    require "i18n"
+    require "i18n/backends/yaml"
+      I18n.backend = I18n::Backends::YAML.new.tap do |backend|
+      backend.load_paths << Dir.current + "/config/locales"
+      backend.load
+    end
+    ```
+
+    ## **Step 4** - create a user migration
+    generate a migration with:
+    ```
+    lucky db.migration AddLanguageToUser
+    ```
+
+    edit the file to look like (of course, the number will vary):
+    ```
+    # db/migrations/20191228100116_add_language_to_user.cr
+    class AddLanguageToUser::V20191228100116 < Avram::Migrator::Migration::V1
+      def migrate
+        alter table_for(User) do
+          add lang : String, default: "en"
+        end
+      end
+
+      def rollback
+        alter table_for(User) do
+          remove :lang
+        end
+      end
+    end
+    ```
+
+    and of course migrate
+    ```
+    lucky db.migrate
+    ```
+
+    ## **Step 5** - update the user model (add the new attribute)
+    ```
+    # src/models/user.cr
+      table do
+        column lang : String
+        column email : String
+        column encrypted_password : String
+       end
+    ```
+
+    ## **Step 6** - Update sign_up_user.cr (to validate the available languages) - ideally at somepoint in the future - lucky will autodect and autocreate the language list based on the config/locales yml files.
+    ```
+    # src/operations/sign_up_user.cr
+    class SignUpUser < User::SaveOperation
+      param_key :user
+      # Change password validations in src/operations/mixins/password_validations.cr
+      include PasswordValidations
+
+      permit_columns email
+      attribute password : String
+      attribute password_confirmation : String
+
+      before_save do
+        validate_uniqueness_of email
+        validate_inclusion_of lang, in: ["en"] # or whatever ymls exist ["en", "de", "fr", "it"]
+        Authentic.copy_and_encrypt password, to: encrypted_password
+      end
+    end
+    ```
+
+    ## **Step 7** - create a translator - could the JS capture be integrated into this class?
+    ```
+    # src/components/translator.cr
+    class Translator
+      getter lang : String
+
+      def initialize(user : User? = nil)
+        @lang = user.try(&.lang) || "en"
+      end
+
+      def t(key : String)
+        I18n.t(key, @lang) # this seems better but not available
```

Comment:
What does this mean "better but not available"? I'm not sure I understand

Reply:
oops - spurious reminders that should be deletee

Reply:
fixed